### PR TITLE
ci/docs: Add mdbook-linkcheck

### LINKED ISSUE #521: Add mdbook-linkcheck to CI
https://github.com/Michael-F-Bryan/mdbook-linkcheck

I've been catching these manually[1], but unsurprisingly computers are better at parsing things than my fallible biological parser.

[1] https://github.com/containers/bootc/pull/520/files/615ead5c9cc85066ef07af1da5321d1c112acc4f#diff-e0b4a488eccbd4e71db5387b0916834520b83da8da89b17f1842760ffd5e1963

### PR DESCRIPTION
**Pull request by jeckersb**
Fixes #521

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -43,7 +51,7 @@ jobs:
       - name: Upload artifact
         uses: actions/upload-pages-artifact@v3
         with:
-          path: ./docs/book
+          path: ./docs/book/html
```
Comment (jeckersb) [inline .github/workflows/docs.yml:54]: This is the key fix since #523

**Comment by jeckersb** [comment]
Workflow in my fork for `68c629c` - https://github.com/jeckersb/bootc/actions/runs/9099106467

Deployed to [my fork pages](https://john.eckersberg.com/bootc/)

**Comment by jeckersb** [comment]
Setting this to draft, looking to incorporate the [previous feedback](https://github.com/containers/bootc/pull/523#issuecomment-2111043918) from @miabbott that I missed until just now.

**Comment by miabbott** [comment]
> Setting this to draft, looking to incorporate the [previous feedback](https://github.com/containers/bootc/pull/523#issuecomment-2111043918) from @miabbott that I missed until just now.

I've been trying to get the right set of conditions so we could just reuse the `docs.yml` jobs, but haven't been successful yet. I'd say just have the duplicated code for now and punt the reuse problem until later.

**Comment by jeckersb** [comment]
> > Setting this to draft, looking to incorporate the [previous feedback](https://github.com/containers/bootc/pull/523#issuecomment-2111043918) from @miabbott that I missed until just now.
> 
> I've been trying to get the right set of conditions so we could just reuse the `docs.yml` jobs, but haven't been successful yet. I'd say just have the duplicated code for now and punt the reuse problem until later.

:+1:  I've pretty much hit the same wall. I'll punt and put it back to ready.

**Comment by jeckersb** [comment]
Hm it's also worth noting that in the [current run](https://github.com/containers/bootc/actions/runs/9099327444/job/25011638412?pr=535), the `CI / docs` job was skipped, even though this PR has the `documentation` label.  Is there a race condition where the jobs can start before the labeler runs, so the `documentation` label hasn't been added in time for the job to pick it up?

**Comment by miabbott** [comment]
> Hm it's also worth noting that in the [current run](https://github.com/containers/bootc/actions/runs/9099327444/job/25011638412?pr=535), the `CI / docs` job was skipped, even though this PR has the `documentation` label. Is there a race condition where the jobs can start before the labeler runs, so the `documentation` label hasn't been added in time for the job to pick it up?

Yeah, I was running into that as well. Been trying to play with `workflow_run` and `pull_request_target`.

Might be easier to gate the `docs.yml` jobs on the path of the files changed? 

```yaml
on:
  pull_request:
    paths:
      - docs/**
      - **.md
```

Fire the build job on any PR or push to `main`, but gate the deploy job on only pushes to `main`?

**Review by cgwalters — APPROVED** [review]

**Comment by cgwalters** [comment]
>  Is there a race condition where the jobs can start before the labeler runs, so the documentation label hasn't been added in time for the job to pick it up?

Right, hmm.  I think we may be able to make them part of the same workflow, and gate later jobs on the labeler too?

> Might be easier to gate the docs.yml jobs on the path of the files changed?

That would work for sure, just at the minor cost of duplicating the docs detection logic.

Alternatively, it's also possible to have workflows react to label changes.  I successfully prototyped that in the past.

**Comment by jeckersb** [comment]
Also worth noting that apparently the label state is "sticky"?  I went into the CI workflow well after this PR had the correct label and forced a [new workflow run](https://github.com/containers/bootc/actions/runs/9099327444/attempts/2?pr=535), and it still skipped the docs job.

**Comment by cgwalters** [comment]
It seems fine to me to also just run the docs check on every PR, it's pretty cheap.  What's less cheap is running all the other CI on docs PRs.

**Review by cgwalters — APPROVED** [review]
I think this is good to merge as is/

**Comment by jeckersb** [comment]
> I think this is good to merge as is, right?

Yeah at the very least it's a step in the right direction.  We can do any tweaks or refactoring in future patches.

**Pull request merged**